core/src/main/java/org/apache/gravitino/credential/CredentialProviderManager.java

```diff
@@ -0,0 +1,72 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.gravitino.credential;
+
+import com.google.common.base.Preconditions;
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CredentialProviderManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CredentialProviderManager.class);
+  private Map<String, CredentialProvider> credentialProviders;
+
+  public CredentialProviderManager() {
+    this.credentialProviders = new ConcurrentHashMap<>();
+  }
+
+  public void registerCredentialProvider(
+      String catalogName, CredentialProvider credentialProvider) {
+    CredentialProvider current = credentialProviders.putIfAbsent(catalogName, credentialProvider);
```

Comment:
Do we support one catalog with multiple credential vendors?  If so, seems this map cannot support it.

Reply:
Not in current PR, I prefer to support it later in integrate Credential vending to Gravitino server to have a better overall design.